lua/pac3/editor/server/wear.lua

```diff
@@ -1,21 +1,35 @@
+local next = next
+local type = type
+local istable = istable
+local IsValid = IsValid
+local tostring = tostring
+
 pace.StreamQueue = pace.StreamQueue or {}
 
-local frame_number = 0
-local last_frame
-local ERROR_COLOR = Color(228, 37, 37)
+local function startQueueTimer()
+	timer.Create("pac_check_stream_queue", 0.1, 0, function()
+		if pace.BusyStreaming then return end
+
+		local item = table.remove(pace.StreamQueue)
+		if not item then return end
 
-local function catchError(err)
-	pac.Message(ERROR_COLOR, 'Error: ', err)
-	pac.Message(debug.traceback())
+		local data = item.data
+		local filter = item.filter
+		local callback = item.callback
+
+		local allowed, reason = pace.SubmitPartNow(data, filter)
+		if isfunction(callback) then
+			callback(allowed, reason)
+		end
+	end)
 end
 
-timer.Create("pac_check_stream_queue", 0.1, 0, function()
-	if not pace.BusyStreaming and #pace.StreamQueue ~= 0 then
-		xpcall(pace.SubmitPart, catchError, unpack(table.remove(pace.StreamQueue)))
+timer.Create("pac_check_stream_queue_timer", 0.2, 0, function()
+	if not timer.Exists("pac_check_stream_queue") then
+		startQueueTimer()
 	end
-
-	frame_number = frame_number + 1
 end)
+startQueueTimer()
```

Comment:
This timer exists solely as a rescue timer, to keep the main queue timer running even if it errors.

Reply:
Couldn't that be solved with xpcall? (I've never seen this pattern before in glua, but I can see some benefits using a rescue timer)

Reply:
Yes, I was trying something new.
I thought it would be more performant, but after some tests, I think simpler is better here.

I went with a `ProtectedCall` solution instead 👍